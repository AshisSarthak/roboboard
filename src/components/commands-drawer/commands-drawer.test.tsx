import { fireEvent, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import CommandsDrawer from "./commands-drawer";

describe("rendersCommandsDrawer", () => {
  test("snapshotTest", () => {
    const { asFragment } = render(<CommandsDrawer />);
    expect(screen.getByText("Enter Command")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test("input value change in commands section", async () => {
    const { asFragment } = render(<CommandsDrawer />);
    const submitBtn = screen.getByRole("button");
    expect(submitBtn).toBeDisabled();
    const input = screen.getByRole("textbox");
    await fireEvent.change(input, { target: { value: "PLACE 0,0,EAST" } });
    expect(submitBtn).not.toBeDisabled();
    fireEvent.click(submitBtn);
  });
});
