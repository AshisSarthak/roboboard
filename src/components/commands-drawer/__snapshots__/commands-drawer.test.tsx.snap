// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendersCommandsDrawer input value change in commands section 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-2ertzu"
  >
    <div
      class="MuiBox-root css-12l3w89"
      role="presentation"
    >
      <header
        class="commands-header"
      >
        Enter Command
      </header>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      />
      <div
        class="commands-body MuiBox-root css-0"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="filled-multiline-flexible"
            id="filled-multiline-flexible-label"
          >
            Tell Robot What to do
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline css-8ewcdo-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <textarea
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
              id="filled-multiline-flexible"
              required=""
              style="height: 0px; text-transform: uppercase; overflow: hidden;"
            />
            <textarea
              aria-hidden="true"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
              readonly=""
              style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); text-transform: uppercase; padding: 0px; width: 100%;"
              tabindex="-1"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  Tell Robot What to do *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <footer
          class="commands-footer"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium Mui-disabled MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
            disabled=""
            tabindex="-1"
            type="button"
          >
            Submit
          </button>
        </footer>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`rendersCommandsDrawer snapshotTest 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-2ertzu"
  >
    <div
      class="MuiBox-root css-12l3w89"
      role="presentation"
    >
      <header
        class="commands-header"
      >
        Enter Command
      </header>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      />
      <div
        class="commands-body MuiBox-root css-0"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="filled-multiline-flexible"
            id="filled-multiline-flexible-label"
          >
            Tell Robot What to do
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline css-8ewcdo-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <textarea
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
              id="filled-multiline-flexible"
              required=""
              style="height: 0px; text-transform: uppercase; overflow: hidden;"
            />
            <textarea
              aria-hidden="true"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
              readonly=""
              style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); text-transform: uppercase; padding: 0px; width: 100%;"
              tabindex="-1"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  Tell Robot What to do *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <footer
          class="commands-footer"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium Mui-disabled MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
            disabled=""
            tabindex="-1"
            type="button"
          >
            Submit
          </button>
        </footer>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
